app/models/discourse_post_event/invitee.rb

```diff
@@ -24,14 +24,18 @@ def self.statuses
     end
 
     def self.create_attendance!(user_id, post_id, status)
-      invitee = Invitee.create!(
-        status: Invitee.statuses[status.to_sym],
-        post_id: post_id,
-        user_id: user_id,
-      )
-      invitee.event.publish_update!
-      invitee.update_topic_tracking!
-      invitee
+      begin
+        invitee = Invitee.create!(
+          status: Invitee.statuses[status.to_sym],
+          post_id: post_id,
+          user_id: user_id,
+        )
+        invitee.event.publish_update!
+        invitee.update_topic_tracking!
+        invitee
+      rescue ActiveRecord::RecordNotUnique
+        # do nothing in case multiple new attendances would be created very fast
+      end
```

Comment:
Method-level rescue blocks ftw! 😃

```suggestion
      invitee = Invitee.create!(
        status: Invitee.statuses[status.to_sym],
        post_id: post_id,
        user_id: user_id,
      )
      invitee.event.publish_update!
      invitee.update_topic_tracking!
      invitee
    rescue ActiveRecord::RecordNotUnique
      # do nothing in case multiple new attendances would be created very fast
```